moneta-core/src/main/java/org/javamoney/moneta/FastMoney.java

```diff
@@ -682,6 +682,9 @@ public FastMoney divide(double divisor) {
         if (NumberVerifier.isInfinityAndNotNaN(divisor)) {
             return new FastMoney(0L, getCurrency());
         }
+        if (divisor == 0.0d) {
+            throw new ArithmeticException("Division by zero");
```

Comment:
Just to be sure: is the message of the exception is the same as for Double? Because there might be some tests who checks the message